internal/kms/keyprotect.go

```diff
@@ -0,0 +1,234 @@
+	KeyProtectRegionKey     = "KP_REGION"
+
+	keyProtectServiceInstanceID = "KP_SERVICE_INSTANCE_ID"
+	// The following options are part of the Kubernetes Secrets.
+	// #nosec:G101, no hardcoded secrets, only configuration keys.
+	KeyProtectServiceAPIKey   = "KP_SERVICE_API_KEY"
+	KeyProtectCustomerRootKey = "KP_CUSTOMER_ROOT_KEY"
+
+	KeyProtectSessionToken = "KP_SESSION_TOKEN"
+	KeyProtectCRK          = "KP_CRK_ARN"
+)
+
+var _ = RegisterProvider(Provider{
+	UniqueID:    kmsTypeKeyProtectMetadata,
+	Initializer: initKeyProtectKMS,
+})
+
+type KeyProtectKMS struct {
+	// basic options to get the secret
+	namespace  string
+	secretName string
+
+	// standard KeyProtect configuration options
+	serviceAPIKey     string
+	customerRootKey   string
+	serviceInstanceID string
+	region            string
+	sessionToken      string
+	client            *kp.Client
+	crk               string
+}
+
+func initKeyProtectKMS(args ProviderInitArgs) (EncryptionKMS, error) {
+	kms := &KeyProtectKMS{
+		namespace: args.Namespace,
+	}
+	// required options for further configuration (getting secrets)
+	err := setConfigString(&kms.secretName, args.Config, KeyProtectSecretNameKey)
+	if errors.Is(err, errConfigOptionInvalid) {
+		return nil, err
+	} else if errors.Is(err, errConfigOptionMissing) {
+		kms.secretName = KeyProtectMetadataDefaultSecretsName
+	}
+
+	err = setConfigString(&kms.serviceInstanceID, args.Config, keyProtectServiceInstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// read the Kubernetes Secret with credentials
+	secrets, err := kms.getSecrets()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secrets for %T: %w", kms,
+			err)
+	}
+
+	err = setConfigString(&kms.serviceAPIKey, secrets, KeyProtectServiceAPIKey)
+	if err != nil {
+		return nil, err
+	}
+	err = setConfigString(&kms.customerRootKey, secrets, KeyProtectCustomerRootKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// KeyProtectSessionToken is optional
+	err = setConfigString(&kms.sessionToken, secrets, KeyProtectSessionToken)
+	if errors.Is(err, errConfigOptionInvalid) {
+		return nil, err
+	}
+
+	// KeyProtect Region is optional
+	err = setConfigString(&kms.region, args.Config, KeyProtectRegionKey)
+	if errors.Is(err, errConfigOptionInvalid) {
+		return nil, err
+	}
+
+	// crk arn is optional
+	err = setConfigString(&kms.crk, secrets, KeyProtectCRK)
+	if errors.Is(err, errConfigOptionInvalid) {
+		return nil, err
+	}
+
+	return kms, nil
+}
+
+func (kms *KeyProtectKMS) getSecrets() (map[string]interface{}, error) {
+	c, err := k8s.NewK8sClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to Kubernetes to "+
+			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
+	}
+
+	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
+		kms.secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w",
+			kms.namespace, kms.secretName, err)
+	}
+
+	config := make(map[string]interface{})
```

Comment:
this should be a string of interfaces? why not string of string?

Reply:
it is the same model we followed for all other kms interfaces integration so this .